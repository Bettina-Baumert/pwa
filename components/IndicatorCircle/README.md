IndicatorCircle Component
---

The IndicatorCircle serves as an animated loading indicator.
It shows a rotating circle in GMD style.

## Getting Started

```
import { IndicatorCircle } from 'Templates/components';

<IndicatorCircle />
<IndicatorCircle size={32} />
<IndicatorCircle color="#fff" />
```

## Props

### size

_Default_: `32` (variables.loadingIndicator.size)

_Type_: `number`  

The size of the component in px.

### color

_Default_: `#5ccee3` (colors.accent)

_Type_: `string`  

The color of the component.
