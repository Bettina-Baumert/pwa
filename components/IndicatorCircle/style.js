/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import cxs from 'cxs';

const spinner = cxs({
  animation: 'rotate 1.6s linear infinite',
  margin: 'auto',
  transformOrigin: 'center center',
  maxWidth: '100%',
  maxHeight: '100%',
});

/**
 * Styles for the rotating circle.
 * @param  {string} color The color of the circle. Default color is "accent".
 * @param  {string} strokeWidth The stroke width of the circle.
 * @return {string} CSS class name.
 */
const circle = (color, strokeWidth) => cxs({
  animation: 'dash 1.2s ease-in-out infinite',
  fill: 'none',
  stroke: color,
  strokeDasharray: '1, 200',
  strokeDashoffset: 0,
  strokeLinecap: 'round',
  strokeMiterlimit: 10,
  strokeWidth,
});

cxs({
  '@keyframes rotate': {
    '100%': { transform: 'rotate(360deg)' },
  },
});

cxs({
  '@keyframes dash': {
    '0%': {
      strokeDasharray: '1, 200',
      strokeDashoffset: '0',
    },
    '50%': {
      strokeDasharray: '89, 200',
      strokeDashoffset: '-35px',
    },
    '100%': {
      strokeDasharray: '89, 200',
      strokeDashoffset: '-124px',
    },
  },
});

export default {
  spinner,
  circle,
};
