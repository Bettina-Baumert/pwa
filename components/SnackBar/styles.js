/*
 *  Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 *  This source code is licensed under the Apache 2.0 license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */

import { css } from 'glamor';

const drawer = css({
  width: '100%',
  background: 'black',
}).toString();

const message = css({
  color: 'white',
}).toString();

export default {
  drawer,
  message,
};
