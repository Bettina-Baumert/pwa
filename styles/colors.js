import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const colors = (process.env.NODE_ENV !== 'test' && themeConfig && themeConfig.colors) ? themeConfig.colors : {};

export default {
  background: '#f8f8f8',
  light: '#fff',
  dark: '#000',
  accent: '#00c3e6',
  placeholder: '#f2f2f2',
  dividers: '#cecece',
  primary: '#fa5400',
  darkGray: '#eaeaea',
  shade3: '#9a9a9a',
  shade4: '#b5b5b5',
  shade5: '#ccc',
  shade6: '#656565',
  shade7: '#eaeaea',
  shade8: '#f7f7f7',
  shade9: '#8d8d8d',
  shade10: '#f4f4f4',
  shade11: '#747474',
  shade12: '#8a8a8f',
  success: '#35cc29',
  warning: '#ff9300',
  error: '#ff0000',
  ...colors,
};
