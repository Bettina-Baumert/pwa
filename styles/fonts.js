import { css } from 'glamor';

const fonts = {
  family: 'system, -apple-system, "SF Pro Display", "Helvetica Neue", "Lucida Grande"',
  rootSize: 17,
  lineHeight: 1.43,
};

css.global('body', {
  font: `${fonts.rootSize}px/${fonts.lineHeight} ${fonts.family}`,
});

export default fonts;
