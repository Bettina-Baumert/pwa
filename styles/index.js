export variables from './variables';
export colors from './colors';
export fonts from './fonts';
export transitions from './transitions';
