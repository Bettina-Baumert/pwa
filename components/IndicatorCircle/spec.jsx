/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow, mount } from 'enzyme';
import IndicatorCircle from './index';
import styles from './style';

describe('<IndicatorCircle />', () => {
  it('should apply the given size', () => {
    const wrapper = shallow(
      <IndicatorCircle size={32} />
    );

    expect(wrapper).toMatchSnapshot();
    const svg = wrapper.find('svg');
    expect(svg.props().width).toBe(32);
    expect(svg.props().height).toBe(32);
  });

  it('should apply the given color', () => {
    const wrapper = mount(
      <IndicatorCircle size={32} color="#fff" />
    );

    expect(wrapper).toMatchSnapshot();
    const correctCssClass = styles.circle('#fff');
    const wrongCssClass = styles.circle('#000');
    const circleHtml = wrapper.find('circle');
    expect(circleHtml.html().indexOf(correctCssClass) >= 0).toBe(true);
    expect(circleHtml.html().indexOf(wrongCssClass) >= 0).toBe(false);
  });
});
