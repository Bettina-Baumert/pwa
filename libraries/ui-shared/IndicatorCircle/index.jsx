import React from 'react';
import PropTypes from 'prop-types';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';
import styles from './style';

/**
 * The circle indicator component.
 * @param {Object} props The component props.
 * @param {Object} props.size Width and height of the circle.
 * @param {Object} props.color Color of the circle.
 * @param {Object} props.strokeWidth Stroke width of the circle.
 * @returns {JSX}
 */
const IndicatorCircle = ({ size, color, strokeWidth }) => (
  <svg
    className={styles.spinner}
    viewBox="25 25 50 50"
    width={size}
    height={size}
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      className={styles.circle(color, strokeWidth)}
      cx="50"
      cy="50"
      r="20"
    />
  </svg>
);

IndicatorCircle.propTypes = {
  color: PropTypes.string,
  size: PropTypes.number,
  strokeWidth: PropTypes.number,
};

IndicatorCircle.defaultProps = {
  color: themeConfig.colors.accent,
  size: themeConfig.variables.loadingIndicator.size,
  strokeWidth: themeConfig.variables.loadingIndicator.strokeWidth,
};

export default IndicatorCircle;
