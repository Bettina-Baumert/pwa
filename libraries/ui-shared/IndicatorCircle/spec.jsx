import React from 'react';
import { shallow, mount } from 'enzyme';
import IndicatorCircle from './index';
import styles from './style';

describe('<IndicatorCircle />', () => {
  it('should apply the given size', () => {
    const wrapper = shallow(<IndicatorCircle size={32} />);

    expect(wrapper).toMatchSnapshot();

    const svg = wrapper.find('svg');

    expect(svg.props().width).toBe(32);
    expect(svg.props().height).toBe(32);
  });

  it('should apply the given color', () => {
    const wrapper = mount(<IndicatorCircle size={32} color="#fff" strokeWidth={4} />);

    expect(wrapper).toMatchSnapshot();

    const correctCssClass = styles.circle('#fff', 4, true);
    const wrongCssClass = styles.circle('#000', 0);
    const circleHtml = wrapper.find('circle');

    expect(circleHtml.html().indexOf(correctCssClass)).toBeGreaterThanOrEqual(0);
    expect(circleHtml.html().indexOf(wrongCssClass)).toBe(-1);
  });
});
