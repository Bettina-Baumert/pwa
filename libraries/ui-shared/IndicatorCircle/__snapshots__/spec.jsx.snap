// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndicatorCircle /> should apply the given color 1`] = `
<IndicatorCircle
  color="#fff"
  paused={true}
  size={32}
  strokeWidth={4}
>
  <svg
    className="css-1llkk3e"
    height={32}
    viewBox="25 25 50 50"
    width={32}
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      className="css-jiweh7"
      cx="50"
      cy="50"
      r="20"
    />
  </svg>
</IndicatorCircle>
`;

exports[`<IndicatorCircle /> should apply the given size 1`] = `
<svg
  className="css-1llkk3e"
  height={32}
  viewBox="25 25 50 50"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className="css-1w8vil2"
    cx="50"
    cy="50"
    r="20"
  />
</svg>
`;
